mama/c_cpp/src/c/mama/publisher.h

```diff
@@ -196,8 +283,61 @@ extern mama_status
 mamaPublisher_getTransport (mamaPublisher   publisher,
                             mamaTransport*  transport);
 
+/**
```

Comment:
Is there any need to have _allocate and _deallocate functions for this? The struct is in a public header so users of mamaPublisherCallbacks can easily allocate it themselves or indeed (more likely) just use a stack allocated struct instead and prevent heap allocation and cleanup altogether (looks like function pointers get copied in the implementations where it's used anyway)